react/features/rnnoise/functions.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import { getJitsiMeetGlobalNS, loadScript } from '../base/util';
+
+let loadRnnoisePromise;
+
+/**
+ * Returns promise that resolves with a RnnoiseProcessor instance.
+ *
+ * @returns {Promise<RnnoiseProcessor>} - Resolves with the blur effect instance.
+ */
+export function createRnnoiseProcessorPromise() {
+    const ns = getJitsiMeetGlobalNS();
+
+    if (ns.effects && ns.effects.createRnnoiseProcessor) {
+        return ns.effects.createRnnoiseProcessor();
+    }
+
+    // Subsequent calls should not attempt to load the script multiple times.
+    if (!loadRnnoisePromise) {
+        loadRnnoisePromise = loadScript('libs/rnnoise-processor.min.js');
+    }
+
+    return loadRnnoisePromise.then(() => ns.effects.createRnnoiseProcessor());
+}
+
+/**
+ * Get the accepted sample length for the rnnoise library. We might want to expose it with flow libdefs.
+ *
+ * @returns {number}
+ */
+export function getSampleLength() {
+    const ns = getJitsiMeetGlobalNS();
+
+    if (ns.effects && ns.effects.RNNOISE_SAMPLE_LENGTH) {
```

Comment:
The optional chaining syntax is cool I'll change the implementation here so It's more clear.
As for using a default value, not sure I agree with that, say for instance we do decide to rename the variable in the rnnoise stream-effects feature and maybe even change it's value, by adding a default alternative at the adaptor level here we would be hiding the fact that the two are no longer linked i.e. we change the value to 960 bcs rnnoise change their implementation and we rename it to RNNOISE_SMPL_LENGTH, the implementation would still use 480 without knowing the wiser. So I'd prefer to keep the error here in case the binding went wrong.
As for when this value could change, like i mentioned above, it's rnnoise specific so they could decide to change it, or we could at some point enhance rnnoise and change the sample rate.